src/renderer/Dataframe.js

```diff
@@ -164,106 +268,129 @@ export default class Dataframe {
-            const scale = diameter / 2 * widthScale;
+            const strokeWidthScale = this._computePointWidthScale(feature, viz);
+
             if (!viz.symbol._default) {
                 const offset = viz.symbolPlacement.eval();
-                center.x += offset[0] * scale;
-                center.y += offset[1] * scale;
+                center.x += offset[0] * strokeWidthScale;
+                center.y += offset[1] * strokeWidthScale;
             }
 
-            const inside = _pointInCircle(p, center, scale);
+            const inside = pointInCircle(p, center, strokeWidthScale);
+
             if (inside) {
                 features.push(this.getFeature(featureIndex));
             }
         }
+
         return features;
     }
 
     _getLinesAtPosition (pos, viz) {
-        return this._getFeaturesFromTriangles(pos, viz.width, viz.filter);
+        return this._getFeaturesFromTriangles('line', pos, viz);
     }
+
     _getPolygonAtPosition (pos, viz) {
-        return this._getFeaturesFromTriangles(pos, viz.strokeWidth, viz.filter);
+        return this._getFeaturesFromTriangles('polygon', pos, viz);
     }
-    _getFeaturesFromTriangles (pos, widthExpression, filterExpression) {
-        const p = wToR(pos.x, pos.y, { center: this.center, scale: this.scale });
+
+    _getFeaturesFromTriangles (geometryType, pos, viz) {
+        const p = wToR(pos.x, pos.y, {
+            center: this.center,
+            scale: this.scale
+        });
+
         const vertices = this.decodedGeom.vertices;
         const normals = this.decodedGeom.normals;
         const breakpoints = this.decodedGeom.breakpoints;
         const features = [];
-        // The viewport is in the [-1,1] range (on Y axis), therefore a pixel is equal to the range size (2) divided by the viewport height in pixels
-        const widthScale = (2 / this.renderer.gl.canvas.clientHeight) / this.scale * this.renderer._zoom;
         // Linear search for all features
         // Tests triangles since we already have the triangulated form
         // Moreover, with an acceleration structure and triangle testing features could be subdivided easily
         let featureIndex = -1;
-        let scale;
-        let computeScale = feature => {
-            // Width is saturated at 336px
-            const width = Math.min(widthExpression.eval(feature), 336);
-            // width is a diameter and scale is radius-like, we need to divide by 2
-            scale = width / 2 * widthScale;
-        };
+        let strokeWidthScale;
+
         for (let i = 0; i < vertices.length; i += 6) {
             if (i === 0 || i >= breakpoints[featureIndex]) {
                 featureIndex++;
                 const feature = this.getFeature(featureIndex);
-                if (this._isFeatureFiltered(feature, filterExpression)) {
+
+                if (geometryType === 'polygon' && !pointInRectangle(p, this._aabb[featureIndex])) {
```

Comment:
Why only polygons?

Reply:
I have been checking this issue closer, because there was a test failing. It failed one checking if the line AABB collided with the viewport:

<img width="375" alt="captura de pantalla 2018-07-24 a las 18 14 07" src="https://user-images.githubusercontent.com/3824953/43152267-6c77b29a-8f6e-11e8-924b-74071bc9398a.png">

As we can see here, `y` point coordinates are _almost_ 0, so I think we can round the coordinates when calculating the point as follows:

```js
export function wToR (x, y, r) {
    return {
        x: ((x - r.center.x) / r.scale).toFixed(2),
        y: ((y - r.center.y) / r.scale).toFixed(2)
    };
}
```

what do you think?